notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
Okay, the main reason I moved the metrics away from Notifier is because I wanted to bring the metrics into scope of AMSet's Sync: https://github.com/prometheus/prometheus/pull/2465/files#diff-4063493f3b3cf719235a092fd180dd56R485

I am not exactly sure how the `Registerer` pattern will help here. Also what do you mean global-free?